src/main.rs

```diff
@@ -1,45 +1,45 @@
 extern crate rand;
 use rand::{Rng,thread_rng};
-use Cardcolor::*;
-use Cardvalue::*;
 
-const COLORS: [Cardcolor; 6] = [Red, Orange, Yellow, Purple, Blue, Green];
-
-// Value Definitions
-#[derive(Clone, Copy)]
-enum Cardcolor {
-    Red,
-    Orange,
-    Yellow,
-    Purple,
-    Blue,
-    Green,
-}
-enum Cardvalue {
-    Common,
-    Bi,
-    Tri,
-    Quad,
-    Quint,
+mod card {
+    #[derive(Clone, Copy)]
+    enum CardColor {
+        Red,
+        Orange,
+        Yellow,
+        Purple,
+        Blue,
+        Green,
+    }
+    enum CardValue {
+        Common,
+        Bi,
+        Tri,
+        Quad,
+        Quint,
+    }
+    struct Card(Color, Value);
```

Comment:
How can you reference just `Color` or `Value`?

Reply:
You can either say `card.0` for `Color` and `card.1` for value, or you can destructure it in a binding, pattern, or function argument as such:
```rust
let Card(color, value) = card;

match card {
    Card(card::Color::Red, value) => do_something_red(value)
    Card(color, value) => do_something_not_red(value)
}

fn print_both(Card(color, value): &Card) { println!("{}, {}", color, value) }
```

Reply:
So Rust recognizes that within `mod Card`, the two `enum`s have `Card` at the beginning?

Reply:
I made a typo there, it should have been `mod card`. The full qualified name of the contents are `card::Card`, `card::Color`, and `card::Value`.